### PR TITLE
Resolve #47

### LINKED ISSUE #47: java.lang.NullPointerException when trying to access AllenUtils.getOntologies().getAncestors()
**Describe the bug**
You get java.lang.NullPointerException when trying to access AllenUtils.getOntologies().getAncestors()

**To Reproduce**
```
from scyjava import jimport
AllenUtils = jimport('sc.fiji.snt.annotation.AllenUtils')
areas = AllenUtils.getOntologies()
for area in areas:
     ancestors = area.getAncestors()
```
gives "java.lang.NullPointerException"

But there is a hint:
if you do:
```
from scyjava import jimport
AllenUtils = jimport('sc.fiji.snt.annotation.AllenUtils')
areas = AllenUtils.getOntologies()
for area in areas:
     **parent = area.getAncestor(1)**
     ancestors = area.getAncestors()
```
no error..

### PR DESCRIPTION
**Pull request by carshadi**
Resolve #47 

- ensure `parentStructure` attribute is initialized by calling `AllenCompartment.getTreePath()`
- fix bug in `AllenCompartment.getChildren(final int level)`
- set pre-computed volume for mouse brain root `OBJMesh`
- `AllenUtils.getOntologies()` returns `List` instead of `Collection`
- create tests for `AllenCompartment` and `AllenUtils`
- misc code cleanup

pinging @tferr

### CODE REVIEW

File: src/main/java/sc/fiji/snt/annotation/AllenCompartment.java [hunk 1]
```diff
@@ -340,7 +336,7 @@ public OBJMesh getMesh() {
 	 */
 	@Override
 	public boolean isParentOf(final BrainAnnotation childCompartment) {
-		if (childCompartment == null || !(childCompartment instanceof AllenCompartment))
```
Comment (tferr) [inline src/main/java/sc/fiji/snt/annotation/AllenCompartment.java:343]: @carshadi , won't you get a NPE here  if the childCompartment being tested is null? nodes in a reconstruction can be associated with a null compartment

File: src/main/java/sc/fiji/snt/annotation/AllenCompartment.java [hunk 2]
```diff
@@ -258,33 +259,29 @@ public AllenCompartment getAncestor(final int level) {
 		final ArrayList<AllenCompartment> children = new ArrayList<>();
 		final Collection<AllenCompartment> allCompartments = AllenUtils.getOntologies();
 		for (AllenCompartment c :  allCompartments) {
-			if (isChildOf(c) && c.getTreePath().size() <= maxLevel)
+			if (c.isChildOf(this) && c.getTreePath().size() <= maxLevel)
```
Comment (tferr) [inline src/main/java/sc/fiji/snt/annotation/AllenCompartment.java:262]: nice! quite embarrassing how this war wrong

**Review by tferr — COMMENTED** [review]
Nice! the tests are going to be super important

File: src/main/java/sc/fiji/snt/annotation/AllenCompartment.java (hunk 1 above)
Comment (carshadi) [reply src/main/java/sc/fiji/snt/annotation/AllenCompartment.java:343]: IntelliJ suggested that the null check was not necessary, apparently `instanceof` returns `false` if the operand is null
https://stackoverflow.com/questions/2950319/is-null-check-needed-before-calling-instanceof

Comment (tferr) [reply src/main/java/sc/fiji/snt/annotation/AllenCompartment.java:343]: I did not know that! good to know. Oh boy. I should adopt IntelliJ. I have never be warned in Eclipse, and have been doing null checks from day one!

**Pull request merged**